Tests/Microsoft.AppCenter.Test.Windows/AppCenterTest.cs

```diff
@@ -515,10 +515,97 @@ public void ParseAppSecretOnePlatformTerminatingSemicolon()
+            var parsedSecret = AppCenter.GetSecretAndTargetForPlatform(secrets, platformId);
+            var expected = $"appSecret={appSecret};target={targetToken}";
+            Assert.AreEqual(expected, parsedSecret);
+        }
+
+        /// <summary>
+		/// Verify throw exception when finding none of the keys.
+		/// </summary>
+		[TestMethod]
+        [ExpectedException(typeof(AppCenterException))]
+        public void CheckWhenFoundNoneOfTheKeys()
+        {
+            var invalidePlatformIdentifier = "invalidePlatformIdentifier";
+            var appSecret = Guid.NewGuid().ToString();
+            var targetToken = Guid.NewGuid().ToString();
+            var platformId = "ios";
+            var secrets = $"{platformId}={appSecret};{platformId}Target={targetToken}";
+            var parsedSecret = AppCenter.GetSecretAndTargetForPlatform(secrets, invalidePlatformIdentifier);
+        }
+
+        /// <summary>
+        /// Verify throw exception when both keys are empty.
+        /// </summary>
+        [TestMethod]
+        [ExpectedException(typeof(AppCenterException))]
+        public void CheckWhenBothKeysAreEmpty()
+        {
+            var appSecret = string.Empty;
+            var targetToken = string.Empty;
+            var platformId = "ios";
+            var secrets = $"{platformId}={appSecret};{platformId}Target={targetToken}";
+            var parsedSecret = AppCenter.GetSecretAndTargetForPlatform(secrets, platformId);
+        }
+
+        /// <summary>
+        /// Verify parse when there is several platforms and both app secret and token.
+        /// </summary>
+        [TestMethod]
+        public void ParseAppSecretAndTargetMultiplePlatform()
+        {
+            var appSecret = Guid.NewGuid().ToString();
+            var anotherAppSecret = Guid.NewGuid().ToString();
+            var targetToken = Guid.NewGuid().ToString();
+            var platformId = "android";
+            var secrets = $"{platformId}={appSecret};ios={anotherAppSecret};{platformId}Target={targetToken};iosTarget={anotherAppSecret}";
+            var parsedSecret = AppCenter.GetSecretAndTargetForPlatform(secrets, platformId);
+            var expected = $"appSecret={appSecret};target={targetToken}";
+            Assert.AreEqual(expected, parsedSecret);
+        }
+
+        /// <summary>
+        /// Verify parse when there is only token
+        /// </summary>
+        [TestMethod]
+        public void ParseTargetTerminatingSemicolon()
+        {
+            var targetToken = Guid.NewGuid().ToString();
+            var platformId = "android";
+            var secrets = $"{platformId}Target={targetToken};";
+            var parsedSecret = AppCenter.GetSecretAndTargetForPlatform(secrets, platformId);
+            var expected = $"target={targetToken}";
+            Assert.AreEqual(expected, parsedSecret);
+        }
+
+        /// <summary>
+        /// Verify parse when there is only token
```

Comment:
The summary of this test is the same as the one above, doesn't describe the test